unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Billboard/Billboard.cs

```diff
@@ -78,25 +60,26 @@ Vector3 GetLookAtVector()
                 // Note (Zak): Model x,y,z are axis that we want to enable/disable
                 // while lookAtDir x,y,z are the components of the look-at vector
                 if (!model.x || model.z)
-                    lookAtDir.y = entityTransform.forward.y;
+                    lookAtDir.y = EntityTransform.forward.y;
                 if (!model.y)
-                    lookAtDir.x = entityTransform.forward.x;
+                    lookAtDir.x = EntityTransform.forward.x;
             }
 
             return lookAtDir.normalized;
         }
 
-        void ChangeOrientation()
+
+        private void Awake()
         {
-            if (entityTransform == null)
-                return;
-            Vector3 lookAtVector = GetLookAtVector();
-            if (lookAtVector != Vector3.zero)
-                entityTransform.forward = lookAtVector;
-        }
+            model = new Model();
+            Tr = transform;
 
-        private void CameraPositionChanged(Vector3 current, Vector3 previous) { ChangeOrientation(); }
+            Environment.i.serviceLocator.Get<IBillboardsController>().BillboardAdded(gameObject);
```

Comment:
It creates a circular dependency between `BillboardsController` and `Billboard` (as you also use Billboard from the controller). In my opinion it's a bad design and dependency management violation.
I am not sure how you approach it here.
Is it possible to call `BillboardsController` from the place you create a billboard?

Reply:
So the proper approach would be via `DataStore`. From `Billboard` you add itself to the DataStore and the controller works with that collection

Reply:
So the proper approach would be via `DataStore`. From `Billboard` you add itself to the DataStore and the controller works with that collection